pyo3-arrow/src/array.rs

```diff
@@ -284,6 +288,57 @@ impl PyArray {
                     .collect::<Vec<_>>();
                 Arc::new(StringViewArray::from(slices))
             }
+            DataType::Timestamp(unit, tz) => {
+                // We normalize all datetimes to datetimes in UTC.
+                let values: Vec<Option<chrono::DateTime<Utc>>> = match tz {
+                    Some(_) => {
+                        let vs: Vec<Option<chrono::DateTime<FixedOffset>>> = obj.extract()?;
+                        vs.into_iter()
+                            .map(|v| v.map(|dt| dt.with_timezone(&Utc)))
+                            .collect()
+                    }
+                    None => {
+                        let vs: Vec<Option<chrono::NaiveDateTime>> = obj.extract()?;
+                        vs.into_iter()
+                            .map(|v| v.map(|naive| naive.and_utc()))
+                            .collect()
+                    }
+                };
+                match unit {
+                    TimeUnit::Second => {
+                        let values: Vec<_> =
+                            values.iter().map(|v| v.unwrap().timestamp()).collect();
+                        Arc::new(TimestampSecondArray::from(values).with_timezone_opt(tz.clone()))
+                    }
+                    TimeUnit::Millisecond => {
+                        let values: Vec<_> = values
+                            .iter()
+                            .map(|v| v.unwrap().timestamp_millis())
+                            .collect();
+                        Arc::new(
+                            TimestampMillisecondArray::from(values).with_timezone_opt(tz.clone()),
+                        )
+                    }
+                    TimeUnit::Microsecond => {
+                        let values: Vec<_> = values
+                            .iter()
+                            .map(|v| v.unwrap().timestamp_micros())
+                            .collect();
+                        Arc::new(
+                            TimestampMicrosecondArray::from(values).with_timezone_opt(tz.clone()),
+                        )
+                    }
+                    TimeUnit::Nanosecond => {
+                        let values: Vec<_> = values
+                            .iter()
+                            .map(|v| v.unwrap().timestamp_nanos_opt().unwrap())
```

Comment:
We should probably the case where nanosecond timestamps overflow and error instead of panic